staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -4115,19 +4184,28 @@ func (dd *DeploymentDescriber) Describe(namespace, name string, describerSetting
 	if err != nil {
 		return "", err
 	}
-	selector, err := metav1.LabelSelectorAsSelector(d.Spec.Selector)
-	if err != nil {
-		return "", err
-	}
+
 	var events *corev1.EventList
 	if describerSettings.ShowEvents {
-		events, _ = dd.client.CoreV1().Events(namespace).Search(scheme.Scheme, d)
+		events, _ = searchEvents(dd.client.CoreV1(), d, describerSettings.ChunkSize)
 	}
 
-	return describeDeployment(d, selector, d.DeepCopy(), events, dd)
+	var oldRSs, newRSs []*appsv1.ReplicaSet
```

Comment:
This is moved out of `describeDeployment`. Doing so allows us to simplify the arguments passed, and seems more in keeping with a pattern I've noticed in this file where `Describe` gathers the data and `describeFoo` formats/prints it.